webknossos/webknossos/dataset/view.py

```diff
@@ -828,31 +1008,53 @@ def get_buffered_slice_reader(
         absolute_bounding_box: Optional[NDBoundingBox] = None,  # in mag1
         use_logging: bool = False,
     ) -> "BufferedSliceReader":
-        """
-        The returned reader yields slices of data along a specified axis.
-        Internally, it reads multiple slices from disk at once and buffers the data.
-
-        Arguments:
-
-        * The user can specify where the writer should start:
-            * `relative_bounding_box` in Mag(1)
-            * `absolute_bounding_box` in Mag(1)
-            * ⚠️ deprecated: `offset` and `size` in the current Mag,
-              `offset` used to be relative for `View` and absolute for `MagView`
-        * `buffer_size`: amount of slices that get buffered
-        * `dimension`: dimension along which the data is sliced
-          (x: `0`, y: `1`, z: `2`; default is `2`)).
+        """Get a buffered reader for efficiently reading data slices.
+
+        Creates a BufferedSliceReader that allows efficient reading of data slices by
+        buffering multiple slices in memory. This is particularly useful when reading
+        large datasets slice by slice.
+
+        Args:
+            buffer_size (int, optional): Number of slices to buffer in memory.
+                Defaults to 32.
+            dimension (int, optional): Dimension along which to slice the data.
+                0 for x, 1 for y, 2 for z. Defaults to 2 (z-axis).
+            relative_offset (Optional[Vec3IntLike], optional): Offset relative to view's
+                position in Mag(1). Defaults to None.
+            absolute_offset (Optional[Vec3IntLike], optional): Absolute offset in Mag(1).
+                Defaults to None.
+            relative_bounding_box (Optional[NDBoundingBox], optional): Bounding box relative
+                to view's position in Mag(1). Defaults to None.
+            absolute_bounding_box (Optional[NDBoundingBox], optional): Absolute bounding box
+                in Mag(1). Defaults to None.
+            use_logging (bool, optional): Whether to log reading operations.
+                Defaults to False.
+
+        Returns:
+            BufferedSliceReader: A reader object that yields data slices.
+
+        Examples:
+            ```python
+            view = layer.get_mag("1").get_view(size=(100, 100, 10))
+
+            # Create a reader with default settings (z-slices)
+            reader = view.get_buffered_slice_reader()
+            for slice_data in reader:
+                process_slice(slice_data)
 
-        The reader must be used as a context manager using the `with` syntax (see example below).
-        Entering the context returns an iterator yielding slices (np.ndarray).
+            # Read y-slices with custom buffer size
+            reader = view.get_buffered_slice_reader(
+                buffer_size=10,
+                dimension=1,  # y-axis
+                relative_offset=(10, 0, 0)
+            )
```

Comment:
The BufferedSliceReader is a context manager. The usage shown in the examples is wrong. It should be used like:

```python
    view = ...
    with view.get_buffered_slice_reader() as reader:
        for slice_data in reader:
            ...
```